### PR TITLE
Reject non-finite prototype-memory hyperparameters at construction

### PR DESCRIPTION
**Pull request by Kayvan-Zahiri**
## Summary
- Relands the useful part of #195: `PrototypeMemoryConfig` now rejects non-finite `bandwidth`, `novelty_threshold`, and `update_rate`.
- Does not rewrite inf-observation softmax into a uniform simplex. Public `predict` stays fail-visible when logits are corrupted, matching the close reason on #195.

## Test plan
- [x] `.venv/bin/python -m pytest tests/test_prototype_memory.py -q`
- [x] `.venv/bin/python -m ruff check alberta_framework/core/prototype_memory.py tests/test_prototype_memory.py`

Made with Cursor. No Slop measured-run receipt (not an approved Codex or Claude Code client).

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by lalalune — APPROVED** [review]
Maintainer review: the claimed defect is real, the current head fixes it without broadening the contract unexpectedly, and the focused regression coverage matches the changed boundary. Approved subject to required checks and integration-order revalidation.

**Review by lalalune — APPROVED** [review]
Maintainer re-review: exact-head diff inspected and focused regression suite passed locally. The claimed boundary defect is real and the implementation is appropriately scoped. Approved for squash merge.
